src/qz/printer/action/WebApp.java

```diff
@@ -117,36 +122,68 @@ public WebApp() {
 
     /** Starts JavaFX thread if not already running */
     public static synchronized void initialize() throws IOException {
-        //JavaFX native libs
-        if (SystemUtilities.isJar() && Constants.JAVA_VERSION.greaterThanOrEqualTo(Version.valueOf("11.0.0"))) {
-            System.setProperty("java.library.path", new File(SystemUtilities.detectJarPath()).getParent() + "/libs/");
-        }
-
         if (instance == null) {
-            new Thread(() -> Application.launch(WebApp.class)).start();
-            startup.set(false);
-        }
+            startupLatch = new CountDownLatch(1);
+
+            // JDK11+ depends bundled javafx
+            if (Constants.JAVA_VERSION.greaterThanOrEqualTo(Version.valueOf("11.0.0"))) {
+                // JavaFX native libs
+                if (SystemUtilities.isJar()) {
+                    System.setProperty("java.library.path", new File(SystemUtilities.detectJarPath()).getParent() + "/libs/");
+                }
+
+                // Monocle default for unit tests
+                boolean useMonocle = true;
+                if (PrintSocketServer.getTrayManager() != null) {
+                    // Honor user override
+                    useMonocle = PrintSocketServer.getTrayManager().isMonoclePreferred();
+                }
+                if (useMonocle) {
+                    log.trace("Initializing monocle platform");
 
-        for(int i = 0; i < (TIMEOUT * 1000); i += SLEEP) {
-            if (startup.get()) { break; }
+                    System.setProperty("javafx.platform", "monocle"); // Standard JDKs
+                    System.setProperty("glass.platform", "Monocle"); // Headless JDKs
+                    System.setProperty("monocle.platform", "Headless");
 
-            log.trace("Waiting for JavaFX..");
-            try { Thread.sleep(SLEEP); } catch(Exception ignore) {}
+                    //software rendering required headless environments
+                    if (PrintSocketServer.isHeadless()) {
+                        System.setProperty("prism.order", "sw");
```

Comment:
Quoting the private bug tracker:

> There is 1 thing that is still open: rendering on Linux with es2 (instead of sw) and Monocle. Should be doable to fix though.
> ...
> rendering on Linux with es2 and Monocle is now working. Need to modify build procedures.

Interestingly, this passed in #586, so I'm not sure what the stability impact of this is to Linux.

Reply:
Gluon explains that this is likely due to the fallback order and that it's very likely it would still work.  Anyway, they provided a new build with this patched, we just need to update the gluon binaries.